tests/test_sync_to_downstream.sh

```
@@ -207,6 +207,202 @@ echo "$hostile_output" | grep -qE "it is a symbolic link|resolves outside MERGEP
+  && fail "non-filtered consumer leaked into output: $sync_out"
+
+# 3) --paths filter restricts target set within the planned PR.
+sync_out=$(MERGEPATH_ROOT_OVERRIDE="$SYNC_MP" "$SCRIPT" "$sha_A" --dry-run --paths "scripts/hooks/the-hook.sh" 2>&1)
+echo "$sync_out" | grep -q "+ scripts/hooks/the-hook.sh" \
+  || fail "--paths filter excluded the requested path"
+echo "$sync_out" | grep -q "+ scripts/coderabbit-wait.sh" \
+  && fail "--paths filter did not exclude scripts/coderabbit-wait.sh"
+
+# 4) Commit that only touches kit + templated → no canonical targets,
+#    summary marks each consumer as ⊘ (skipped, deferred-only).
+deferred_only_sha=$(git -C "$SYNC_MP" rev-list HEAD --reverse | sed -n '2p')  # commit A
```

Shellcheck (0.11.0)

[warning] deferred_only_sha appears unused. Verify use (or export if used externally).

(SC2034)